src/plugins/embeddable/public/lib/embeddables/embeddable.tsx

```diff
@@ -131,6 +132,28 @@ export abstract class Embeddable<
     return this.output;
   }
 
+  public async getExplicitInputIsEqual(lastInput: Partial<TEmbeddableInput>): Promise<boolean> {
+    const currentInput = this.getExplicitInput();
```

Comment:
I think I don't understand the implementation of this method, need more context :D 

Why don't we just deepEqual two explicit inputs here?

Reply:
Some keys in the generic embeddable input need special comparisons. Doing a deep equal runs into some issues, so I figured it was better to omit the special cases and diff them explicitly rather than write a new deep equal function. 

Here are the issues this solves in the generic input case:

**title**
- A title which is a blank string must be considered _distinct_ from a title which is undefined because an undefined title means 'fall back to the saved object title' and a blank string means 'hide this title'.
- An input with the title key undefined must be considered _equal_ to an input with no title key at all. The deepEqual considers these cases distinct
Instead of deep equal, we use `===` on the title to capture all these cases.

**hidePanelTitles**
- An input with `hidePanelTitles` set to false should be considered equal to an input with hidePanelTitles set to undefined **or** an input where the `hidePanelTitles` key is missing.
Instead of deep equal, we use `===` on `hidePanelTitles` cast to a boolean to ensure these cases are maintained.

Reply:
Got this part, thanks for the details, 

More specific questions: 

1. This method operates only on `explicitInput`? Both `lastInput` and `currentInput` are actually `lastExplicitInput` and `currentExplicitInput`? If yes, then I'd name them as such
2. Why do we split those objects into genericEmbeddableInputIsEqual and nonGeneric import instead of just breaking up `title` and `hidePanelTitles`?

Reply:
1. Good call, I have renamed them!
2. The separation of all generic input into its own method is so downstream applications (like Maps in this PR) don't have to worry about diffing or separating out _any_ of the generic input. Having it as a separate method that each implementation can optionally call is the best way I could think of to get this to work. I was considering using a `super.getIsExplicitInputEqual(lastInput)` call instead, but then there would be a conflict with the generic implementation in the abstract `embeddable.tsx` where it would diff _all_ of the input.

Reply:
Do we really need the concept of "generic" input? I'm thinking the two cases mentioned above don't need it.

> **title**
> - A title which is a blank string must be considered _distinct_ from a title which is undefined because an undefined > title means 'fall back to the saved object title' and a blank string means 'hide this title'.
> - An input with the title key undefined must be considered _equal_ to an input with no title key at all. The deepEqual considers these cases distinct
Instead of deep equal, we use `===` on the title to capture all these cases.

For **title**, we just need to use a deep comparison function which handles `undefined` and a missing key the same.

> **hidePanelTitles**
> - An input with `hidePanelTitles` set to false should be considered equal to an input with hidePanelTitles set to undefined **or** an input where the `hidePanelTitles` key is missing.
Instead of deep equal, we use `===` on `hidePanelTitles` cast to a boolean to ensure these cases are maintained.

Do we really want to treat `hidePanelTitles` value `false` as `undefined`, maybe in this case it is OK that the deep comparison find the inequality.